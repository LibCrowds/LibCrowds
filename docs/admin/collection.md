# Collection Admin

This section contains a series of pages via which users with admin rights can
manage the platform's collection microsites.

## Open Collection

Here you can select the collection microsite that you wish to manage. Click
'Open' to display the all additional settings pages for a microsite.

## New Collection

To create a new collection specify a name and description, then click 'Create'.

Following creation you will be given access to a range of additional
configuration options, which are described in more detail below.

## Settings

This is where the core settings for the microsite are configured, each of which
are explained below.

| Setting            | Description
|--------------------|------------------------------------------------------------------------------------------------|
| Published          | Once published, a microsite is added to the platform homepage and is accessible to all users.  |
| Name               | Applies branding throughout the collection microsite.                                          |
| Short name         | Used as an identifier and to form the microsite endpoint (e.g. `/collection/short_name`).      |
| Tagline            | Appears over the hero image on the microsite's homepage and forms a part of it's title.        |
| Description        | Appearson the microsite's homepage and forms its meta description.                             |
| Background Image   | The URL for an image used to provide a consistent theme for all of the microsite's core pages. |
| Logo               | The URL for a logo to be shown alongside the name on the microsite's homepage.                 |
| Forum URL          | The URL for the Discuss link in the microsite's main navigation bar.                           |
| Task Presenter     | The module used to present tasks for all of the microsite's projects.                          |
| Data Reuse License | The license by which all data collected via the microsite is made available.                   |

## Content

A set of markdown editors are provided here for modifying content for the About,
Projects and Data pages. Changes are made in real-time, although you may have
to refresh your browser to see them.

For the About page, navigation items will be generated from each level two
header, created in Markdown by using the ## tag, for example:

```
## Contact

This is the contact details section.
```

These navigation items can be clicked to scroll the users window to the top
of that section.

For more guidance on the GitHub flavoured markdown used, see this
[GitHub Guide](https://guides.github.com/features/mastering-markdown/).

## Terminology

You can use the options provided here to modify the domain object terminology
used throughout the collection microsite. For instance, task could be renamed
to objective, or task run to contribution.

## Tags

Tags are used to help users more easily location the types of project within a
collection that they're interested in. For each tag type created via this
admin section a new filter will be added to the microsite's Contribute
page (among others).

![Collection Admin - Tags](../assets/admin-collection-tags.png?raw=true)

To add a new tag click the 'Add tag' button at the top of the table. You
will then be given options to choose a tag type (e.g. Location), a tag name
(e.g. London) and a colour. Note that the colour applies to all tags of the
same type.

Once tags are created they can be added to individual projects via the
[Project Admin](admin/project.md) section.

Tags can be edited or removed by locating them in the table and clicking the
corresponding buttons. If you want to edit the color for a particular tag type
just click edit on any tags of that type.

## Featured

Here you can set the projects that appear on the Featured Projects section of
the collection microsite's homepage.

![Collection Admin - Featured](../assets/admin-collection-featured.jpg?raw=true)

## Delete

Here you can permanently delete a collection microsite.
