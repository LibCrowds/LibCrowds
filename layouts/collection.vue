<template>
  <div id="collection-base-layout">
    <div id="collection-background" :style="bgStyle"></div>
    <collection-navbar
      :collection-config="collectionConfig">
    </collection-navbar>
    <main>
      <nuxt
        :collection-config="collectionConfig">
      </nuxt>
    </main>
    <app-footer></app-footer>

    <notifications position="bottom left"></notifications>

  </div>
</template>

<script>
import localConfig from '@/local.config'
import CollectionNavbar from '@/components/navs/CollectionNavbar'
import AppFooter from '@/components/footers/App'

export default {
  head () {
    const tmpl = `%s - ${this.collectionConfig.name} | ${localConfig.brand}`
    return {
      titleTemplate: tmpl,
      meta: [
        {
          hid: 'description',
          name: 'description',
          content: this.collectionConfig.description
        },

        // Facebook Open Graph Markup
        { property: 'og:url', content: window.location.href },
        { property: 'og:title', content: this.collectionConfig.tagline },
        {
          property: 'og:description',
          content: this.collectionConfig.description
        }
      ]
    }
  },

  components: {
    CollectionNavbar,
    AppFooter
  },

  computed: {
    bgStyle () {
      const url = this.collectionConfig.bgImg
      return {
        background: `url(${url}) no-repeat center left /cover fixed`
      }
    },
    collectionConfig () {
      const collectionName = this.$route.params.collectionname
      return localConfig.collections[collectionName]
    }
  }
}
</script>

<style lang="scss" scoped>
#collection-base-layout {
  flex: 1 1 auto;
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  #collection-background {
    z-index: -1;
    position: fixed;
    top: 0;
    left: 0;
    min-height: 100%;
    height: auto;
    width: 100%;
    min-width: 1024px;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
  }

  main {
    flex: 1 1 auto
  }
}
</style>
