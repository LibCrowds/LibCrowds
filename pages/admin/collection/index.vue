<template>
  <card-base :title="title" help="Manage the collection microsites">
    <b-btn
      slot="controls"
      variant="success"
      class="float-right"
      size="sm"
      :to="{
        name: 'admin-collection-new'
      }">
      New
    </b-btn>

    <infinite-loading-table
      ref="table"
      :fields="tableFields"
      no-border
      domain-object="category">
      <template slot="action" scope="category">
        <b-btn
          variant="success"
          size="sm"
          block
          exact
          :to="{
            name: 'admin-collection-short_name-settings',
            params: {
              short_name: category.item.short_name
            }
          }">
          Open
        </b-btn>
      </template>
    </infinite-loading-table>

  </card-base>
</template>

<script>
import PybossaForm from '@/components/forms/PybossaForm'
import CardBase from '@/components/cards/Base'
import InfiniteLoadingTable from '@/components/tables/InfiniteLoading'

export default {
  layout: 'admin-collection-dashboard',

  middleware: 'is-admin',

  data () {
    return {
      title: 'Open Collection',
      tableFields: {
        name: {
          label: 'Name'
        },
        created: {
          label: 'Created',
          class: 'text-center d-none d-md-table-cell',
          sortable: true
        }
      }
    }
  },

  head () {
    return {
      title: this.title
    }
  },

  components: {
    PybossaForm,
    CardBase,
    InfiniteLoadingTable
  },

  beforeMount () {
    this.$store.dispatch('UPDATE_CURRENT_COLLECTION', {})
  }
}
</script>
