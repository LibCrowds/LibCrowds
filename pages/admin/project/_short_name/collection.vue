<template>
  <card-base :title="title" :description="description">

    <b-form-input
      slot="controls"
      v-model="filter"
      class="search-control"
      size="sm"
      :placeholder="`Type to search by ${filterBy}`">
    </b-form-input>

    <infinite-loading-table
      ref="table"
      :fields="tableFields"
      :filter="filter"
      :filter-by="filterBy"
      domain-object="category">
      <template slot="action" scope="collection">
        <b-btn
          variant="success"
          size="sm"
          :disabled="collection.item.id == project.category_id"
          @click="setCollection(collection.item)">
          Select
        </b-btn>
      </template>
    </infinite-loading-table>
  </card-base>
</template>

<script>
import { notifications } from '@/mixins/notifications'
import { fetchProjectByName } from '@/mixins/fetchProjectByName'
import { metaTags } from '@/mixins/metaTags'
import InfiniteLoadingTable from '@/components/tables/InfiniteLoading'
import CardBase from '@/components/cards/Base'

export default {
  layout: 'admin-project-dashboard',

  mixins: [ fetchProjectByName, notifications, metaTags ],

  data () {
    return {
      title: 'Collection',
      description: 'Choose the collection that this project belongs to.',
      tableFields: {
        name: {
          label: 'Name',
          sortable: true
        },
        created: {
          label: 'Created',
          class: 'text-center d-none d-md-table-cell',
          sortable: true
        }
      },
      searchParams: {},
      updatedCollectionId: null,
      filter: null,
      filterBy: 'name'
    }
  },

  components: {
    InfiniteLoadingTable,
    CardBase
  },

  computed: {
    project () {
      return this.$store.state.currentProject
    }
  },

  methods: {
    /**
     * Set the collection for the project.
     * @param {Object} collection
     *   The collection.
     */
    setCollection (collection) {
      this.$axios.$put(`/api/project/${this.project.id}`, {
        category_id: collection.id
      }).then(data => {
        this.notifySuccess({ message: `Project moved to ${collection.name}` })
        this.$store.dispatch('UPDATE_CURRENT_PROJECT', data)
      }).catch(err => {
        this.$nuxt.error(err)
      })
    }
  }
}
</script>
