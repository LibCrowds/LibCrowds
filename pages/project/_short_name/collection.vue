<template>
  <b-card
    no-body>
    <div slot="header" class="mb-0">
      <h6 class="mb-0">{{ title }}</h6>
      <p class="text-muted mb-0">
        <small>
          Choose the collection microsite that this project should appear on.
        </small>
      </p>
    </div>

    <infinite-loading-table
      ref="table"
      :fields="tableFields"
      no-border
      domain-object="category">
      <template slot="action" scope="collection">
        <b-btn
          variant="success"
          size="sm"
          block
          :disabled="collection.item.id == project.category_id"
          @click="setCollection(collection.item)">
          Select
        </b-btn>
      </template>
    </infinite-loading-table>
  </b-card>
</template>

<script>
import { notifications } from '@/mixins/notifications'
import { fetchProjectByName } from '@/mixins/fetchProjectByName'
import InfiniteLoadingTable from '@/components/tables/InfiniteLoading'

export default {
  layout: 'project-dashboard',

  mixins: [ fetchProjectByName, notifications ],

  data () {
    return {
      title: 'Set Collection',
      tableFields: {
        name: {
          label: 'Name',
          sortable: true
        },
        created: {
          label: 'Created',
          class: 'text-center d-none d-md-table-cell',
          sortable: true
        }
      },
      searchParams: {},
      updatedCollectionId: null
    }
  },

  head () {
    return {
      title: this.title
    }
  },

  components: {
    InfiniteLoadingTable
  },

  computed: {
    project () {
      return this.$store.state.currentProject
    }
  },

  methods: {
    /**
     * Set the collection for the project.
     * @param {Object} collection
     *   The collection.
     */
    setCollection (collection) {
      this.$axios.$put(`/api/project/${this.project.id}`, {
        category_id: collection.id
      }).then(data => {
        this.notify({
          type: 'success',
          title: 'Success',
          message: `Project moved to ${collection.name}`
        })
        this.$store.dispatch('UPDATE_CURRENT_PROJECT', data)
      }).catch(err => {
        this.$nuxt.error({ statusCode: err.statusCode, message: err.message })
      })
    }
  }
}
</script>
