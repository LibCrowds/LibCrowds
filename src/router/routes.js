import config from '@/config'
import store from '@/store'

import Core from '@/components/layouts/Core'
import Landing from '@/pages/Landing'

import About from '@/pages/pybossa/home/About'
import Data from '@/pages/pybossa/home/Data'
import TOS from '@/pages/pybossa/help/TOS'
import Privacy from '@/pages/pybossa/help/Privacy'
import Cookies from '@/pages/pybossa/help/Cookies'
import Api from '@/pages/pybossa/help/Api'

export default [

  // Core site
  { path: '/', name: 'core', component: Core },
  { path: 'tos', name: 'tos', component: TOS },
  { path: 'privacy', name: 'privacy', component: Privacy },
  { path: 'cookies', name: 'cookies', component: Cookies },
  { path: 'api', name: 'api', component: Api },

  // PYBOSSA meta-category sites
  {
    path: '/site/:shortname',
    component: {
      render (c) {
        return c('router-view')
      }
    },
    beforeEnter: (to, from, next) => {
      let siteConfig = config.sites.find(function (site) {
        return site.shortname === to.params.shortname
      })

      if (typeof siteConfig === 'undefined') {
        next({ name: '404' })
      }

      store.commit('SET_SITE_CONFIG', siteConfig)
      next()
    },
    children: [
      { path: '', name: 'home', component: Landing },
      { path: 'about', name: 'about', component: About },
      { path: 'data', name: 'data', component: Data }
    ]
  },

  // Errors
  {
    path: '/404',
    name: '404',
    alias: '*',
    component: require('@/pages/Error'),
    props: {
      title: 'Page Not Found',
      description: 'We could not find the page you were looking for'
    }
  }
]
