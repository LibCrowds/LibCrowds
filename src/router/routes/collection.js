import siteConfig from '@/siteConfig'
import CollectionBase from '@/pages/collection/Base'
import CollectionHome from '@/pages/collection/Home'
import CollectionAbout from '@/pages/collection/About'
import CollectionData from '@/pages/collection/Data'
// import CollectionResults from '@/pages/collection/Results'
import CollectionContribute from '@/pages/collection/Contribute'
import CollectionPresenter from '@/pages/collection/Presenter'

const routes = [
  {
    path: '/collection/:collectionname',
    component: CollectionBase,
    beforeEnter: function (to, from, next) {
      if (to.params.collectionname in siteConfig.collections) {
        next()
      } else {
        next({ name: '404' })
      }
    },
    children: [
      {
        path: '/',
        name: 'collection-home',
        component: CollectionHome
      },
      {
        path: 'about',
        name: 'collection-about',
        component: CollectionAbout
      },
      {
        path: 'data',
        name: 'collection-data',
        component: CollectionData
      },
      {
        path: 'contribute',
        name: 'collection-contribute',
        component: CollectionContribute
      },
      {
        path: 'project/:shortname/presenter',
        name: 'collection-presenter',
        component: CollectionPresenter
      }
    ]
  }
]

// Conditional routes (TODO: Update)
// if (config.resultsComponent) {
//   routes.push({
//     path: '/collection/:collectionname/statistics',
//     name: 'results',
//     component: Collection
//   })
// }

export default routes
