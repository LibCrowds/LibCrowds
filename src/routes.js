import config from '@/config'

export default [
  {
    path: '/',
    component: require('@/pages/Landing')
  },
  {
    path: '/site/:sitename',
    component: require('@/components/layouts/Main'),
    beforeEnter: (to, from, next) => {
      if (!(to.params.sitename in config.pybossaSites)) {
        next({ name: '404' })
      }
      next()
    },
    children: [
      {
        path: '',
        name: 'home',
        component: require('@/pages/Landing')
      },
      {
        path: 'about',
        name: 'about',
        component: require('@/pages/pybossa/home/About')
      },
      {
        path: 'projects',
        name: 'projects',
        component: require('@/pages/pybossa/projects/Projects')
      },
      {
        path: 'stats',
        name: 'stats',
        component: require('@/pages/pybossa/home/Stats')
      },
      {
        path: 'results',
        name: 'results',
        component: require('@/pages/pybossa/home/Results')
      },
      {
        path: 'data',
        name: 'data',
        component: require('@/pages/pybossa/home/About')
      }
    ]
  },
  {
    path: '/404',
    name: '404',
    alias: '*',
    component: require('@/pages/Error'),
    meta: { title: 'Page Not Found' }
  }
]
