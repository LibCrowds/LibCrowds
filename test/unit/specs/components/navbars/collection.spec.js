import Vuex from 'vuex'
import VueRouter from 'vue-router'
import BootstrapVue from 'bootstrap-vue'

import { routes } from '@/test/fixtures/routes'
import collection from '@/test/fixtures/collection.json'
import pbTestResponses from '@/test/fixtures/pybossaTestResponses.json'
import { mount, createLocalVue } from 'vue-test-utils'
import CollectionNavbar from '@/components/navbars/Collection'

describe('Collection navbar', () => {
  let localVue = null
  let wrapper = null
  let store = null
  let router = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    localVue.use(Vuex)
    localVue.use(VueRouter)
    store = new Vuex.Store({
      state: {
        currentUser: {}
      }
    })
    router = new VueRouter({
      routes
    })
  })

  it('renders correctly when signed out', () => {
    wrapper = mount(CollectionNavbar, {
      localVue,
      router,
      propsData: {
        collection: collection,
        currentUser: {}
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('renders correctly when signed in as non-admin', () => {
    const user = pbTestResponses.getAccount.data.user
    user.admin = false
    wrapper = mount(CollectionNavbar, {
      localVue,
      store,
      router,
      propsData: {
        collection: collection,
        currentUser: JSON.parse(JSON.stringify(user))
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('renders correctly when signed in as admin', () => {
    const user = pbTestResponses.getAccount.data.user
    user.admin = true
    wrapper = mount(CollectionNavbar, {
      localVue,
      store,
      router,
      propsData: {
        collection: collection,
        currentUser: JSON.parse(JSON.stringify(user))
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
