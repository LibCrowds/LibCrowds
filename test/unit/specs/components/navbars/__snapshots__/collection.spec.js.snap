// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection navbar renders correctly when signed in as admin 1`] = `

<nav id="collection-navbar"
     data-server-rendered="true"
     class="navbar navbar-dark navbar-expand-md"
>
  <div class="container">
    <a href="#/collection/playbills"
       target="_self"
       class="navbar-brand"
    >
      <span>
        In the Spotlight
      </span>
    </a>
    <button type="button"
            data-toggle="collapse"
            data-target="#main-nav-collapse"
            aria-controls="main-nav-collapse"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="
        hamburger
        hamburger--collapse
        navbar-toggler
        navbar-toggler-right"
    >
      <span class="hamburger-box">
        <span class="hamburger-inner">
        </span>
      </span>
    </button>
    <div id="main-nav-collapse"
         class="navbar-collapse collapse"
         style="display:none;"
    >
      <ul class="nav navbar-nav">
        <li class="nav-item">
          <a href="#/collection/playbills/about"
             target="_self"
             class="nav-link"
          >
            About
          </a>
        </li>
        <li class="nav-item">
          <a href="#/collection/playbills/projects"
             target="_self"
             class="nav-link"
          >
            Contribute
          </a>
        </li>
        <li class="nav-item">
          <a href="http://example.com/discuss"
             target="_self"
             aria-disabled="false"
             class="nav-link"
          >
            Discuss
          </a>
        </li>
        <li class="nav-item">
          <a href="#/collection/playbills/data"
             target="_self"
             class="nav-link"
          >
            Data
          </a>
        </li>
      </ul>
      <ul class="nav navbar-nav">
        <div id="announcements-list">
          <button id="announcements-toggle"
                  type="button"
                  class="nav-link d-none d-lg-flex flex-row btn btn-secondary"
          >
            <icon name="bell">
            </icon>
            <!---->
          </button>
          <nuxt-link to="[object Object]"
                     class="d-lg-none nav-link"
          >
            Notifications
          </nuxt-link>
          <span style="display:none;">
            <div id="announcements-container"
                 class="dropdown-menu-right d-none d-md-block card"
            >
              <div class="card-header">
                <div>
                  Notifications
                </div>
              </div>
              <div id="announcements">
              </div>
              <no-ssr>
                <infinite-loading class="infinite-loading">
                  <span slot="no-results">
                    <span>
                      No announcements have been made yet
                    </span>
                  </span>
                  <span slot="no-more">
                    <!---->
                  </span>
                </infinite-loading>
              </no-ssr>
            </div>
          </span>
        </div>
        <li class="nav-item b-nav-dropdown dropdown">
          <a href="#"
             aria-haspopup="true"
             aria-expanded="false"
             class="nav-link dropdown-toggle"
          >
            <span>
              JohnDoe
            </span>
          </a>
          <div role="menu"
               class="dropdown-menu dropdown-menu-right"
          >
            <a href="#/account/JohnDoe"
               role="menuitem"
               target="_self"
               class="dropdown-item"
            >
              Profile
            </a>
            <a href="#/account/JohnDoe/settings/profile"
               role="menuitem"
               target="_self"
               class="dropdown-item"
            >
              Settings
            </a>
            <span>
              <div role="separator"
                   class="dropdown-divider"
              >
              </div>
              <a href="#/admin/project/open"
                 role="menuitem"
                 target="_self"
                 class="dropdown-item"
              >
                Project Admin
              </a>
              <a href="#/"
                 role="menuitem"
                 target="_self"
                 class="dropdown-item active"
              >
                Collection Admin
              </a>
              <a href="#/admin/site/dashboard"
                 role="menuitem"
                 target="_self"
                 class="dropdown-item"
              >
                Site Admin
              </a>
            </span>
            <div role="separator"
                 class="dropdown-divider"
            >
            </div>
            <a role="menuitem"
               href="#"
               target="_self"
               aria-disabled="false"
               class="dropdown-item"
            >
              Sign Out
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>

`;

exports[`Collection navbar renders correctly when signed in as non-admin 1`] = `

<nav id="collection-navbar"
     data-server-rendered="true"
     class="navbar navbar-dark navbar-expand-md"
>
  <div class="container">
    <a href="#/collection/playbills"
       target="_self"
       class="navbar-brand"
    >
      <span>
        In the Spotlight
      </span>
    </a>
    <button type="button"
            data-toggle="collapse"
            data-target="#main-nav-collapse"
            aria-controls="main-nav-collapse"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="
        hamburger
        hamburger--collapse
        navbar-toggler
        navbar-toggler-right"
    >
      <span class="hamburger-box">
        <span class="hamburger-inner">
        </span>
      </span>
    </button>
    <div id="main-nav-collapse"
         class="navbar-collapse collapse"
         style="display:none;"
    >
      <ul class="nav navbar-nav">
        <li class="nav-item">
          <a href="#/collection/playbills/about"
             target="_self"
             class="nav-link"
          >
            About
          </a>
        </li>
        <li class="nav-item">
          <a href="#/collection/playbills/projects"
             target="_self"
             class="nav-link"
          >
            Contribute
          </a>
        </li>
        <li class="nav-item">
          <a href="http://example.com/discuss"
             target="_self"
             aria-disabled="false"
             class="nav-link"
          >
            Discuss
          </a>
        </li>
        <li class="nav-item">
          <a href="#/collection/playbills/data"
             target="_self"
             class="nav-link"
          >
            Data
          </a>
        </li>
      </ul>
      <ul class="nav navbar-nav">
        <div id="announcements-list">
          <button id="announcements-toggle"
                  type="button"
                  class="nav-link d-none d-lg-flex flex-row btn btn-secondary"
          >
            <icon name="bell">
            </icon>
            <!---->
          </button>
          <nuxt-link to="[object Object]"
                     class="d-lg-none nav-link"
          >
            Notifications
          </nuxt-link>
          <span style="display:none;">
            <div id="announcements-container"
                 class="dropdown-menu-right d-none d-md-block card"
            >
              <div class="card-header">
                <div>
                  Notifications
                </div>
              </div>
              <div id="announcements">
              </div>
              <no-ssr>
                <infinite-loading class="infinite-loading">
                  <span slot="no-results">
                    <span>
                      No announcements have been made yet
                    </span>
                  </span>
                  <span slot="no-more">
                    <!---->
                  </span>
                </infinite-loading>
              </no-ssr>
            </div>
          </span>
        </div>
        <li class="nav-item b-nav-dropdown dropdown">
          <a href="#"
             aria-haspopup="true"
             aria-expanded="false"
             class="nav-link dropdown-toggle"
          >
            <span>
              JohnDoe
            </span>
          </a>
          <div role="menu"
               class="dropdown-menu dropdown-menu-right"
          >
            <a href="#/account/JohnDoe"
               role="menuitem"
               target="_self"
               class="dropdown-item"
            >
              Profile
            </a>
            <a href="#/account/JohnDoe/settings/profile"
               role="menuitem"
               target="_self"
               class="dropdown-item"
            >
              Settings
            </a>
            <!---->
            <div role="separator"
                 class="dropdown-divider"
            >
            </div>
            <a role="menuitem"
               href="#"
               target="_self"
               aria-disabled="false"
               class="dropdown-item"
            >
              Sign Out
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>

`;

exports[`Collection navbar renders correctly when signed out 1`] = `

<nav id="collection-navbar"
     data-server-rendered="true"
     class="navbar navbar-dark navbar-expand-md"
>
  <div class="container">
    <a href="#/collection/playbills"
       target="_self"
       class="navbar-brand"
    >
      <span>
        In the Spotlight
      </span>
    </a>
    <button type="button"
            data-toggle="collapse"
            data-target="#main-nav-collapse"
            aria-controls="main-nav-collapse"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="
        hamburger
        hamburger--collapse
        navbar-toggler
        navbar-toggler-right"
    >
      <span class="hamburger-box">
        <span class="hamburger-inner">
        </span>
      </span>
    </button>
    <div id="main-nav-collapse"
         class="navbar-collapse collapse"
         style="display:none;"
    >
      <ul class="nav navbar-nav">
        <li class="nav-item">
          <a href="#/collection/playbills/about"
             target="_self"
             class="nav-link"
          >
            About
          </a>
        </li>
        <li class="nav-item">
          <a href="#/collection/playbills/projects"
             target="_self"
             class="nav-link"
          >
            Contribute
          </a>
        </li>
        <li class="nav-item">
          <a href="http://example.com/discuss"
             target="_self"
             aria-disabled="false"
             class="nav-link"
          >
            Discuss
          </a>
        </li>
        <li class="nav-item">
          <a href="#/collection/playbills/data"
             target="_self"
             class="nav-link"
          >
            Data
          </a>
        </li>
      </ul>
      <ul class="nav navbar-nav">
        <ul class="nav navbar-nav">
          <li class="nav-item">
            <a href="#/account/signin?next=%2F"
               target="_self"
               class="nav-link"
            >
              Sign in
            </a>
          </li>
          <li class="nav-button nav-item">
            <a href="#/account/register"
               target="_self"
               class="nav-link"
            >
              Sign up
            </a>
          </li>
        </ul>
      </ul>
    </div>
  </div>
</nav>

`;
